cloudnet-wrapper-jvm/src/main/java/de/dytanic/cloudnet/wrapper/Wrapper.java

```diff
@@ -2279,75 +2275,27 @@ private boolean startApplication() throws Exception {
         ApplicationEnvironmentEvent preStartApplicationEvent = new ApplicationEnvironmentEvent(this, this.config.getServiceConfiguration().getServiceId().getEnvironment());
         this.eventManager.callEvent(preStartApplicationEvent);
 
-        File[] files = this.workDirectory.listFiles();
-        if (files == null) {
-            this.logger.log(LogLevel.ERROR,
-                    "",
-                    new FileNotFoundException("no files found!"));
-            return false;
-        }
-
-        File entry = null;
-        ServiceEnvironment environment = null;
-
-        for (ServiceEnvironment subEnvironment : preStartApplicationEvent.getEnvironmentType().getEnvironments()) {
-            if (entry != null) {
-                break;
-            }
+        File entry = new File(commandLineArguments.get(0));
 
-            entry = this.find(files, file -> {
-                String lowerName = file.getName().toLowerCase();
-                return file.exists() && !file.isDirectory() && lowerName.endsWith(".jar") &&
-                        lowerName.contains(subEnvironment.getName().toLowerCase());
-            });
+        Optional<ServiceEnvironment> environment = Arrays.stream(preStartApplicationEvent.getEnvironmentType().getEnvironments())
+                .filter(env -> entry.getName().toLowerCase().contains(env.getName().toLowerCase()))
+                .findFirst();
 
-            environment = subEnvironment;
-        }
-
-        try {
-            return this.startApplication0(entry, environment);
-        } catch (Throwable throwable) {
-            throwable.printStackTrace(System.err);
+        if (entry.exists() && environment.isPresent()) {
+            return this.startApplication0(entry, environment.get());
+        } else {
+            return false;
```

Comment:
How will the user know, that there's no environment the service can use? Please log an exception for this case.

Reply:
already defined in JVMCloudService. Removing the ApplicationEnvironmentEvent will make this unnecessary, so please say your opinion about that

Reply:
also every errors which are thrown are walking into main method (with the "throws" in method call)